ci/src/datakit_ci.mli

```diff
@@ -0,0 +1,664 @@
+      project. [tests] is a list tests to apply to branches, tags and
+      open PRs within the project. [dashboards] (default
+      [["master"]]) is a list of branches to display in the main
+      dashboard area. *)
+
+  val v: web_config:Web.config -> projects:project list -> t
+  (** [v ~web_config ~project] describe a complete project
+      configuration. *)
+end
+
+val run: ?info:Cmdliner.Term.info -> Config.t Cmdliner.Term.t -> unit
+(** [run ?info config] runs DataKitCI. [info] defaults to a term that
+    describes the binary as [DataKitCI], but does not include any of
+    the other metadata such as versioning. Call {!Cmdliner.Term.info}
+    directly to obtain an [info] value that exposes all this extra
+    data on the resulting command line. *)
+
+val logs: Live_log.manager
+(** The singleton log manager. *)
+
+(** {1:extensions Extensions} *)
+
+module Process: sig
+
+  (** {1:process Extension of Lwt_process}
+
+      Convenience wrappers around [Lwt_process]. *)
+
+  val run_with_exit_status:
+    ?switch:Lwt_switch.t -> ?log:Live_log.t -> ?cwd:string ->
+    ?env:string array -> output:(string -> unit) ->
+    ?log_cmd:Lwt_process.command -> Lwt_process.command ->
+    Unix.process_status Lwt.t
+  (** Run [cmd], passing each chunk of output it produces on stdout or
+      stderr to [output]. A copy of the output is also streamed to
+      our stdout. Returns the exit status of the process when
+      completed. If [log_cmd] is given, it is displayed in all log
+      messages instead of [cmd].  This is useful to hide secret
+      tokens, etc. *)
+
+  val run:
+    ?switch:Lwt_switch.t -> ?log:Live_log.t -> ?cwd:string ->
+    ?env:string array -> output:(string -> unit) ->
+    ?log_cmd:Lwt_process.command -> Lwt_process.command ->
+    unit Lwt.t
+  (** Run [cmd], passing each chunk of output it produces on stdout or
+      stderr to [output]. A copy of the output is also streamed to
+      our stdout. Raises an exception if the process doesn't return
+      an exit status of zero. *)
+
+  val check_status: Lwt_process.command -> Unix.process_status -> unit
+  (** [check_status cmd status] checks that [status] is a successful
+      exit status.  If not, it raises an exception giving [cmd] as the
+      cause. *)
+end
+
+module Monitored_pool: sig
+
+  (** {1:pool Monitored Pool of Workers} *)
+
+  type t
+  (** The type for pools of workers. *)
+
+  val create: string -> int -> t
+  (** [create s n] create a new pool with [n] workers. *)
+
+  val use: t -> ?log:Live_log.t -> ?label:string -> job_id ->
+    (unit -> 'a Lwt.t) -> 'a Lwt.t
+  (** [use t job fn] evaluates [fn ()] with one pool resource held.
+      [job] (and [label]) will be displayed as the reason why the
+      resource is in use. If [log] is provided then a message will be
+      logged if we have to wait, and if the log is cancellable then
+      the user will be able to cancel the operation. *)
+end
+
+module Git: sig
```

Comment:
Note: this was separate to demonstrate how to write a plugin using only the public API.